tests/test_authz_authorizer.py

```diff
@@ -1,130 +1,241 @@
+        assert authz.allow == {ALL: ALL}
+        assert authz.deny == {}
+        assert authz.outcome == DENY
+        assert authz.allowed_resource is None
+        assert authz.allowed_resource is None
+        assert authz.resource == "test_resource"
+        assert authz.permission == "permission_name"
+
+    def test__repr__(self, full_access_auth_header) -> None:
+        authz = self._make_authorizer(jwt=full_access_auth_header)
+        assert repr(authz) == (
             "Authorizer(auth_jwt=<jwt>, "
             "resource_name='test_resource', "
             "permission_name='permission_name')"
         )
 
-    def test__set_policy_w_scope(self):
-        self.authz._set_policy(  # pylint: disable=protected-access
-            "", {"allow": "Lambda", "deny": "Lambda"}
+    def test_validate_one_with_expired(self, full_access_authz_payload) -> None:
+        expired_timestamp = int((datetime.utcnow() - timedelta(seconds=1)).timestamp())
+        with pytest.raises(Unauthorized):
+            self._make_authorizer(
+                token_payload={
+                    **full_access_authz_payload,
+                    "exp": expired_timestamp,
+                }
+            )
+
+
+# pylint: disable=too-many-public-methods
+class TestAuthorizerWithMockedJWT:
+    @staticmethod
+    def _make_mocked_authorizer(token_payload: dict) -> Authorizer:
+        with patch("lbz.authz.authorizer.decode_jwt", lambda _: token_payload):
+            return Authorizer("xx", "test_resource", "permission_name")
+
+    def test_wrong_jwt_authz_payload_raises_permission_denied(self) -> None:
+        with pytest.raises(PermissionDenied):
+            self._make_mocked_authorizer({})
+
+    def test_check_deny_res(self, full_access_authz_payload) -> None:
+        authz = self._make_mocked_authorizer(full_access_authz_payload)
+        authz.deny = {"test_resource": ALL}
+        with pytest.raises(PermissionDenied):
+            authz._check_deny()  # pylint: disable=protected-access
+        assert authz.outcome == DENY
+
+    def test_check_allow_all(self, full_access_authz_payload) -> None:
+        authz = self._make_mocked_authorizer(full_access_authz_payload)
+        authz._check_allow_and_set_resources()  # pylint: disable=protected-access
+        assert authz.outcome == ALLOW
+
+    def test_check_deny_any(self, full_access_authz_payload) -> None:
+        authz = self._make_mocked_authorizer(full_access_authz_payload)
+        authz.deny = {ALL: ALL}
+        with pytest.raises(PermissionDenied):
+            authz._check_deny()  # pylint: disable=protected-access
+        assert authz.outcome == DENY
+
+    def test_check_access_check_deny_outcome(self) -> None:
+        authz = self._make_mocked_authorizer({"allow": {}, "deny": {ALL: ALL}})
+        with pytest.raises(PermissionDenied):
+            authz.check_access()
+
+    def test_check_access_outcome_deny(self, full_access_authz_payload) -> None:
```

Comment:
You have two new tests here:
- test_check_access_check_deny_outcome
- test_check_access_outcome_deny

Unfortunately, they are named almost the same, which doesn't help to understand what you wanted to achieve and won't help to develop that in the future 😉  As I mentioned before, please, you longer, self-descriptive names for the tests.